File: week2/csharp/main/model/Product.cs
namespace csharp.model
{
    public class Product : IIdentifiable<string>
    {
        public string BarCode { get; set; }
        public string Description { get; set; }
        public double Price { get;set; }
        public int Count { get; set; }

        public Product()
        {
            BarCode = string.Empty;
            Description = string.Empty;
            Price = 0.00;
            Count = 0;
        }

        public Product(string id, string description, double price, int count)
        {
            BarCode = id;
            Description = description;
            Price = price;
            Count = count;
        }

        public string getId()
        {
            return BarCode;
        }

        public void setId(string id)
        {
            BarCode = id;
        }

        public override string ToString()
        {
            return $"|{BarCode}| {Description}; {Price}$; {Count} pieces;";
        }
    }
}

File: week2/gradle-java/src/main/java/model/Product.java
package model;

import java.io.Serializable;

public class Product implements Identifiable<String>, Serializable {
    private String barCode;
    private String description;
    private double price;
    private int count;

    public Product() {
        this.barCode = "";
        this.description = "";
        this.price = 0.00;
        this.count = 0;
    }

    public Product(String barCode, String description, double price, int count) {
        this.barCode = barCode;
        this.description = description;
        this.price = price;
        this.count = count;
    }

    public String getId(){
        return barCode;
    }

    @Override
    public void setId(String id) {
        this.barCode = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    @Override
    public String toString(){
        return "|" + barCode + "| "+ description+"; "+price+"$; "+count+" pieces;";
    }
}
